### PR TITLE
improve GetProfile (single call instead of two, plus error handling and private profiles)

### PR DESCRIPTION
**Pull request by mipak**
- instead of sending two requests, the required data is already in the first response which is currently used to get only the rest_id
- currently private profiles return 403 and no data, even there is data returned by twitter
- get error handling to check if profile was not found or suspended (or other reasons)
- added tests to for all of above

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,4 +1,4 @@
-module github.com/n0madic/twitter-scraper
+module github.com/dataxpe/twitter-scraper
```
Comment (n0madic) [inline go.mod:1]: Why?

**Review by n0madic — CHANGES_REQUESTED** [review]
Please fix go.mod

File: go.mod (hunk 1 above)
Comment (mipak) [reply go.mod:1]: just ignore that change. it was necessary so I could use my fork in my code.

Comment (n0madic) [reply go.mod:1]: how can I ignore it in a pull request?

**Pull request closed without merging**